url6be695d833_nikoloz_pachuashvili_Patterneria_tgz-pJ8-org_patterneria_MainJ8/src/org/patterneria/factory/DirectorB.java

```diff
@@ -0,0 +1,42 @@
+package org.patterneria.factory;
+
+/**
+ * Director "B" (Concrete factory)
+ */
+public class DirectorB  implements TheatreDirector{
+
+    /**
+     * Gets the name of director
+     *
+     * @return the name of director
+     */
+    @Override
+    public String getName() {
+        return "B";
+    }
+
+    /**
+     *  Stage the spectacle
+     *
+     * @param play The name of play
+     *
+     * @return The spectacle
+     */
+    @Override
+    public Spectacle stage(String play) {
+
+        Spectacle spectacle;
+
+        if (play == "The 39 Steps") {
+            spectacle = new Les39MarchesByB();
+        }
+        else if(play == "Richard III"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.